detection_rules/packaging.py

```diff
@@ -356,6 +362,13 @@ def rule_count(rule_dict):
 
         return summary_str, changelog_str
 
+    def generate_attack_navigator(self, path: Path) -> Dict[Path, Navigator]:
+        """Generate ATT&CK navigator layer files."""
```

Comment:
because most of the documentation (and typing, defaults, etc) can be derived from existing components, if we feel the need for docs, we could leverage that route. There are som existing libs :
https://pypi.org/project/mkdocs-click/ 
https://sphinx-click.readthedocs.io/en/latest/usage/

or we could roll our own.

We should punt that to a dedicated issue though